vita3k/regmgr/src/regmgr.cpp

```diff
@@ -271,49 +268,57 @@ static void init_system_dreg(RegMgrState &regmgr, const std::wstring pref_path)
 
     for (const auto &reg : reg_template) {
         for (const auto &entry : reg.second) {
-            regmgr.system_dreg[reg.first][entry.name] = entry.value;
+            const std::vector<char> default_value(entry.value.begin(), entry.value.end());
+            auto &value = regmgr.system_dreg[reg.first][entry.name];
+            value.resize(entry.size);
+            value = default_value;
         }
     }
 
     save_system_dreg(regmgr, pref_path);
 }
 
-static std::string fix_category(const std::string& category) {
+static std::string fix_category(const std::string &category) {
     return category.back() == '/' ? category : category + '/';
 }
 
-void set_bin_value(RegMgrState &regmgr, const std::wstring &pref_path, const std::string &category, const std::string &name, const void *buf, const uint32_t size) {
+void set_bin_value(RegMgrState &regmgr, const std::wstring &pref_path, const std::string &category, const std::string &name, const void *buf, const uint32_t bufSize) {
     std::lock_guard<std::mutex> lock(regmgr.mutex);
 
-    const char *data = reinterpret_cast<const char *>(buf); 
-    const std::vector<char> bin(data, data + size);    
-    regmgr.system_dreg[fix_category(category)][name] = std::string(bin.begin(), bin.end());
+    const char *data = reinterpret_cast<const char *>(buf);
+    regmgr.system_dreg[fix_category(category)][name].assign(data, data + bufSize);
 
     save_system_dreg(regmgr, pref_path);
 }
 
 int32_t get_int_value(RegMgrState &regmgr, const std::string &category, const std::string &name) {
     std::lock_guard<std::mutex> lock(regmgr.mutex);
 
-    return std::stoi(regmgr.system_dreg[fix_category(category)][name]);
+    const auto sys = regmgr.system_dreg[fix_category(category)][name];
```

Comment:
```suggestion
    const auto& sys = regmgr.system_dreg[fix_category(category)][name];
```

Reply:
for why ?? value is just read, no need & 🤔

Reply:
No sur of utility but done

Reply:
@Zangetsu38 The original code takes extra execution time to allocate memory for the sys variable and copy its value. Macdu's modification uses a reference, which means that the variable will only be read, so it doesn't need to be copied like the original.

Reply:
oh, i see